openmetadata-service/src/main/java/org/openmetadata/service/apps/bundles/insights/workflows/costAnalysis/processors/AggregatedCostAnalysisReportDataProcessor.java

```diff
@@ -0,0 +1,293 @@
+                frequentlyUsedDataAssets
+                    .getCount()
+                    .withThreeDays(
+                        (double) frequentlyUsedDataAssets.getCount().getThreeDays() + 1));
+            frequentlyUsedDataAssets.withSize(
+                frequentlyUsedDataAssets
+                    .getSize()
+                    .withThreeDays(
+                        (double) frequentlyUsedDataAssets.getSize().getThreeDays() + tableSize));
+          }
+          // Compute 7Days
+          periodTimestamp =
+              TimestampUtils.getEndOfDayTimestamp(TimestampUtils.subtractDays(timestamp, 7));
+
+          if (lifeCycle.getAccessed().getTimestamp() <= periodTimestamp) {
+            unusedDataAssets.withCount(
+                unusedDataAssets
+                    .getCount()
+                    .withSevenDays((double) unusedDataAssets.getCount().getSevenDays() + 1));
+            unusedDataAssets.withSize(
+                unusedDataAssets
+                    .getSize()
+                    .withSevenDays((double) unusedDataAssets.getSize().getSevenDays() + tableSize));
+          } else {
+            frequentlyUsedDataAssets.withCount(
+                frequentlyUsedDataAssets
+                    .getCount()
+                    .withSevenDays(
+                        (double) frequentlyUsedDataAssets.getCount().getSevenDays() + 1));
+            frequentlyUsedDataAssets.withSize(
+                frequentlyUsedDataAssets
+                    .getSize()
+                    .withSevenDays(
+                        (double) frequentlyUsedDataAssets.getSize().getSevenDays() + tableSize));
+          }
+          // Compute 14Days
+          periodTimestamp =
+              TimestampUtils.getEndOfDayTimestamp(TimestampUtils.subtractDays(timestamp, 14));
+
+          if (lifeCycle.getAccessed().getTimestamp() <= periodTimestamp) {
+            unusedDataAssets.withCount(
+                unusedDataAssets
+                    .getCount()
+                    .withFourteenDays((double) unusedDataAssets.getCount().getFourteenDays() + 1));
+            unusedDataAssets.withSize(
+                unusedDataAssets
+                    .getSize()
+                    .withFourteenDays(
+                        (double) unusedDataAssets.getSize().getFourteenDays() + tableSize));
+          } else {
+            frequentlyUsedDataAssets.withCount(
+                frequentlyUsedDataAssets
+                    .getCount()
+                    .withFourteenDays(
+                        (double) frequentlyUsedDataAssets.getCount().getFourteenDays() + 1));
+            frequentlyUsedDataAssets.withSize(
+                frequentlyUsedDataAssets
+                    .getSize()
+                    .withFourteenDays(
+                        (double) frequentlyUsedDataAssets.getSize().getFourteenDays() + tableSize));
+          }
+          // Compute 30Days
+          periodTimestamp =
```

Comment:
these Compute X days look super repetitive, not sure if we can have a loop/method to handle them?

Reply:
I thought about it but we did not touch the CostAnalysis or WebAnalytics schemas and flows.

Maybe there's a hacky way of matching the number of days to the specific attribute getters and setters but I didn't think it was worth the hussle since we should refactor the way we deal with Cost Analysis to make it a bit more flexible as well afaik.